src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -215,13 +215,14 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
 
             if (parameterCustomAttributes.OfType<IFromRouteMetadata>().FirstOrDefault() is { } routeAttribute)
             {
-                factoryContext.TrackedParameters.Add(parameter.Name, RequestDelegateFactoryConstants.RouteAttribute);
-                if (factoryContext.RouteParameters is { } routeParams && !routeParams.Contains(parameter.Name, StringComparer.OrdinalIgnoreCase))
+                var routeName = routeAttribute.Name ?? parameter.Name;
+                factoryContext.TrackedParameters.Add(routeName, RequestDelegateFactoryConstants.RouteAttribute);
```

Comment:
```suggestion
                factoryContext.TrackedParameters.Add(parameter.Name, RequestDelegateFactoryConstants.RouteAttribute);
```

I think the parameter name is preferred for the TrackedParameters list